Welcome to the much improved Quizlet style program!  All you have to do is hit 'run all cells' and you will be prompted for the right stuff

In [2]:
import json
import urllib.request
import random
from difflib import SequenceMatcher

In [3]:
index = input("what is the quizlet set number? \t")

what is the quizlet set number? 	124815649


Sample quizlet ID: 124815649

I suggest hardcoding your client_id into the next cell.  It won't authenticate without a valid one.

In [ ]:
client_id = ''

In [5]:
data = json.load(urllib.request.urlopen('https://api.quizlet.com/2.0/sets/{}?client_id={}&whitespace=1'.format(index, client_id)))
pairs = [(x['term'], x['definition'].lower()) for x in data['terms']]

In [15]:
def quizme(term_pairs):
    right, wrong = 0, 0 #tracking
    random.shuffle(term_pairs) #picks a random ordering of the cards
    for term, ans in term_pairs: 
        user_ans = input(term + ':\t').lower()
        if user_ans == '999': #escape code
            break
        user_ans_parts = user_ans.split(',')
        ans_parts = ans.split(',')
        
        if user_ans == ans:
            print('\nGood job!!')
            right += 1
            continue
        
        else:
            rights = []
            for part in ans_parts:
                for upart in user_ans_parts:
                    if SequenceMatcher(None, part, upart).ratio() > 0.8:
                        rights.append(True)
            if len(rights) == len(ans_parts):    
                print('\nClose enough.  I had it as ' + '\033[92m' + ans + '\033[0m.')
                right += 1
                continue 
        print('\nSorry, the answer was ' + '\033[91m' + ans + '\033[0m.')
        wrong +=1
    print('\nYou got {} questions right and {} questions wrong.'.format(right, wrong))

In [ ]:
quizme(pairs)

GHANA:	a

Sorry, the answer was accra.
CONGO REPUBLIC:	brazzaville

Good job!!
COTE D'IVOIRE:	abidan

Sorry, the answer was abidjan, yamoussoukro.
EQUATORIAL GUINEA:	malabi

Close enough.  I had it as malabo.
ANGOLA:	luanda

Good job!!
NAMIBIA:	windhoek

Good job!!
SOUTH SUDAN:	juba

Good job!!
SIERRA LEONE:	freetown

Good job!!
ERITREA:	asmara

Good job!!
ZIMBABWE:	hare

Sorry, the answer was harare.
MAURITIUS:	port louis

Good job!!
SOUTH AFRICA:	capetown, bloemfontein, pretoria

Close enough.  I had it as cape town, bloemfontein, pretoria.
BURUNDI:	999

You got 9 questions right and 3 questions wrong.


In [9]:
print('Here are the answers')
print(pairs)

Here are the answers
[('SOUTH AFRICA', 'cape town, bloemfontein, pretoria'), ('MADAGASCAR', 'antananarivo'), ('GUINEA BISSAU', 'bissau'), ('BURUNDI', 'bujumbura'), ('CAMEROON', "yaounde'"), ("COTE D'IVOIRE", 'abidjan, yamoussoukro'), ('SWAZILAND', 'mbabane, lobamba'), ('NAMIBIA', 'windhoek'), ('SOUTH SUDAN', 'juba'), ('GAMBIA', 'banjul'), ('ZIMBABWE', 'harare'), ('EQUATORIAL GUINEA', 'malabo'), ('BENIN', 'porto-novo'), ('CHAD', "n'djamena"), ('ERITREA', 'asmara'), ('MAURITIUS', 'port louis'), ('GHANA', 'accra'), ('ETHIOPIA', 'addis ababa'), ('MALI', 'bamako'), ('CONGO REPUBLIC', 'brazzaville'), ('TANZANIA', 'dar es salaam, dodoma'), ('SIERRA LEONE', 'freetown'), ('ANGOLA', 'luanda')]
